### ISSUE TITLE
KeyError: 'ConcreteDamageCityScapes is not in the dataset registry'

### ISSUE THREAD

**Issue by tanveer6715** [open-mmlab/mmsegmentation#1665]
Hi. I have create my own dataset and train the **SegFormer** successfully but when I check the inference speed  of **SegFormer** using **benchmark.py** script it shows the following error as attached. I guess mmsegmentation do not support custom dataset or models to find inference speed or model parameters.
![error](https://user-images.githubusercontent.com/84553134/173306037-1c3162d4-a4d3-4153-b55c-c8319f7f2dfb.PNG)

**Comment by tanveer6715** [comment]
> Hi, could you train and test normally with your checkpoints trained from ConcreteDamageCityScapes datasets?

Yes it works well with dist_test.sh but shows same error when testing with single GPU. I also find one another thing during testing. The result are different when evaluating during training and evaluating using **dist_test.sh**. Will you kindly explain the reason behind it?

**Comment by MengzhangLI** [comment]
> 

Could you use **dist_test.sh with 1 GPU** to inference?

**Comment by tanveer6715** [comment]
> > 
> 
> Could you use **dist_test.sh with 1 GPU** to inference?
> 
> I think it should be unrelated with single or multiple GPU, and you can checkout your training log to see whether `ConcreteDamageCityScapes` is registered in MMCV, theoretically it should be registered like below:
> 
> https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/datasets/isprs.py#L6-L7

Yes **dist_test.sh** with 1 GPU also works. The **ConcreteDamageCityScapes** is already registered that's why it get successfully trained. But I experienced using custom data show an error with **benchmark.py** script and also mmsegmentation do not support custom model to find flops or parameters of model.

**Comment by MengzhangLI** [comment]
> 

The dataset in `benchmark.py` is built [here](https://github.com/open-mmlab/mmsegmentation/blob/master/tools/benchmark.py#L59), which would use [here](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/datasets/builder.py#L85), when I check out the registry, they are all registered like below:

![image](https://user-images.githubusercontent.com/24582831/173764076-84d52e3e-5b48-4e6b-8760-684420d616c7.png)

So could you debug what's your situation here? I think this bug is not related with single/multiple GPUs, let alone `benchmark.py`.

Best,

**Comment by tanveer6715** [comment]
I dont know what causes this problem. It seems to be very old bug or issue when using any custom module. Please check this issue. It is same like mine but I couldnt find any proper solution.
https://github.com/open-mmlab/mmdetection/issues/3751

**Comment by MengzhangLI** [comment]
> I dont know what causes this problem. It seems to be very old bug or issue when using any custom module. Please check this issue. It is same like mine but I couldnt find any proper solution. [open-mmlab/mmdetection#3751](https://github.com/open-mmlab/mmdetection/issues/3751)

I opened this link and found it has been fixed. You could try to take solutions mentioned in this link.

**Comment by tanveer6715** [comment]
> > I dont know what causes this problem. It seems to be very old bug or issue when using any custom module. Please check this issue. It is same like mine but I couldnt find any proper solution. [open-mmlab/mmdetection#3751](https://github.com/open-mmlab/mmdetection/issues/3751)
> 
> I opened this link and found it has been fixed. You could try to take solutions mentioned in this link.

Yes I tried the proposed solutions but it still create problem when testing the custom data.

**Comment by zwbx** [comment]
I also encounter the same bug using single gpu testing, while multiple GPU testing works works well.
I am sure that I have checked the link, and the customer dataset have been registered.

**Comment by WonderNoot** [comment]
I am experiencing the same problem, the dataset configs are correctly imported in the mmdet.datasets module but are somehow not available in the registry when firing up training. I am using an extension of the CocoDataset and even that one is not available. Creating a custom dataset cfg on top of the CocoDataset type results in the same problem.
